src/service/resources-impl.js

```diff
@@ -1272,6 +1313,23 @@ export class Resources {
       }
     }
   }
+
+  /**
+   * Removes the task resource from the discovered first viewport
+   * resources array.
+   * @param {!TaskDef} task
+   */
+  checkForFirstViewportElementsWhenTaskDone_(task) {
```

Comment:
@dvoytenko can i just hook into when the first `layoutComplete_` is called in the `Resource` class? (instead of hooking into the task complete code, this way i can revert my changes)

as for `getResourcesInViewport`, this will either have to be an async method that returns a promise (since previously we collected the resources until we knew the dom was ready) or the performance class will have to call it at right time (ondomready).

let me know if you want to take this offline, thanks!